File: CastIt.Infrastructure/Models/ServerAppSettings.cs
using CastIt.Domain.Enums;
using CastIt.GoogleCast.Enums;

namespace CastIt.Infrastructure.Models
{
    public class ServerAppSettings
    {
        public string FFmpegPath { get; set; }
        public string FFprobePath { get; set; }

        public string CurrentAppMigration { get; set; }
        public bool StartFilesFromTheStart { get; set; }
        public bool PlayNextFileAutomatically { get; set; }
        public bool ForceVideoTranscode { get; set; }
        public bool ForceAudioTranscode { get; set; }
        public VideoScaleType VideoScale { get; set; }
        public bool EnableHardwareAcceleration { get; set; }

        public SubtitleFgColorType CurrentSubtitleFgColor { get; set; }
        public SubtitleBgColorType CurrentSubtitleBgColor { get; set; }
        public SubtitleFontScaleType CurrentSubtitleFontScale { get; set; }
        public TextTrackFontStyleType CurrentSubtitleFontStyle { get; set; }
        public TextTrackFontGenericFamilyType CurrentSubtitleFontFamily { get; set; }
        public double SubtitleDelayInSeconds { get; set; }
        public bool LoadFirstSubtitleFoundAutomatically { get; set; }

        public ServerAppSettings Copy()
        {
            return new ServerAppSettings
            {
                FFmpegPath = FFmpegPath,
                FFprobePath = FFprobePath,

                CurrentAppMigration = CurrentAppMigration,
                StartFilesFromTheStart = StartFilesFromTheStart,
                PlayNextFileAutomatically = PlayNextFileAutomatically,
                ForceVideoTranscode = ForceVideoTranscode,
                ForceAudioTranscode = ForceAudioTranscode,
                VideoScale = VideoScale,
                EnableHardwareAcceleration = EnableHardwareAcceleration,

                CurrentSubtitleFgColor = CurrentSubtitleFgColor,
                CurrentSubtitleBgColor = CurrentSubtitleBgColor,
                CurrentSubtitleFontScale = CurrentSubtitleFontScale,
                CurrentSubtitleFontStyle = CurrentSubtitleFontStyle,
                CurrentSubtitleFontFamily = CurrentSubtitleFontFamily,
                SubtitleDelayInSeconds = SubtitleDelayInSeconds,
                LoadFirstSubtitleFoundAutomatically = LoadFirstSubtitleFoundAutomatically
            };
        }

        public ServerAppSettings UpdateWith(ServerAppSettings other)
        {
            //TODO: FFMPEG ?

            StartFilesFromTheStart = other.StartFilesFromTheStart;
            PlayNextFileAutomatically = other.PlayNextFileAutomatically;
            ForceVideoTranscode = other.ForceVideoTranscode;
            ForceAudioTranscode = other.ForceAudioTranscode;
            VideoScale = other.VideoScale;
            EnableHardwareAcceleration = other.EnableHardwareAcceleration;

            CurrentSubtitleFgColor = other.CurrentSubtitleFgColor;
            CurrentSubtitleBgColor = other.CurrentSubtitleBgColor;
            CurrentSubtitleFontScale = other.CurrentSubtitleFontScale;
            CurrentSubtitleFontStyle = other.CurrentSubtitleFontStyle;
            CurrentSubtitleFontFamily = other.CurrentSubtitleFontFamily;
            SubtitleDelayInSeconds = other.SubtitleDelayInSeconds;
            LoadFirstSubtitleFoundAutomatically = other.LoadFirstSubtitleFoundAutomatically;
            return this;
        }
    }
}

File: CastIt.Android/lib/domain/models/server_app_settings.dart
import 'package:castit/domain/enums/enums.dart';
import 'package:castit/domain/enums/subtitle_bg_color_type.dart';
import 'package:castit/domain/enums/text_track_font_generic_family_type.dart';
import 'package:json_annotation/json_annotation.dart';

part 'server_app_settings.g.dart';

@JsonSerializable()
class ServerAppSettings {
  VideoScaleType get videoScaleType => getVideoScaleType(videoScale);

  SubtitleFgColorType get currentSubtitleFgColorType => SubtitleFgColorType.values[currentSubtitleFgColor];

  SubtitleBgColorType get currentSubtitleBgColorType => SubtitleBgColorType.values[currentSubtitleBgColor];

  SubtitleFontScaleType get currentSubtitleFontScaleType => getSubtitleFontScaleType(currentSubtitleFontScale);

  TextTrackFontStyleType get currentSubtitleFontStyleType => TextTrackFontStyleType.values[currentSubtitleFontStyle];

  TextTrackFontGenericFamilyType get currentSubtitleFontFamilyType => TextTrackFontGenericFamilyType.values[currentSubtitleFontFamily];

  final String fFmpegExePath;
  final String fFprobeExePath;

  final bool startFilesFromTheStart;
  final bool playNextFileAutomatically;
  final bool forceVideoTranscode;
  final bool forceAudioTranscode;
  final int videoScale;
  final bool enableHardwareAcceleration;

  final int currentSubtitleFgColor;
  final int currentSubtitleBgColor;
  final int currentSubtitleFontScale;
  final int currentSubtitleFontStyle;
  final int currentSubtitleFontFamily;
  final double subtitleDelayInSeconds;
  final bool loadFirstSubtitleFoundAutomatically;

  ServerAppSettings({
    required this.fFmpegExePath,
    required this.fFprobeExePath,
    required this.startFilesFromTheStart,
    required this.playNextFileAutomatically,
    required this.forceVideoTranscode,
    required this.forceAudioTranscode,
    required this.videoScale,
    required this.enableHardwareAcceleration,
    required this.currentSubtitleFgColor,
    required this.currentSubtitleBgColor,
    required this.currentSubtitleFontScale,
    required this.currentSubtitleFontStyle,
    required this.currentSubtitleFontFamily,
    required this.subtitleDelayInSeconds,
    required this.loadFirstSubtitleFoundAutomatically,
  });

  factory ServerAppSettings.fromJson(Map<String, dynamic> json) => _$ServerAppSettingsFromJson(json);

  Map<String, dynamic> toJson() => _$ServerAppSettingsToJson(this);
}
